src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -549,42 +549,65 @@ class PostgresEntityDataQueryService(
     }
 
     fun deleteEntityData(
-            entitySetId: UUID, entityKeyIds: Set<UUID>, authorizedPropertyTypes: Map<UUID, PropertyType>
+            entitySetId: UUID,
+            entityKeyIds: Set<UUID>,
+            authorizedPropertyTypes: Map<UUID, PropertyType>,
+            partitionsInfo: PartitionsInfo = partitionManager.getEntitySetPartitionsInfo(entitySetId)
     ): WriteEvent {
-        val connection = hds.connection
-        val numUpdates = connection.use {
-            authorizedPropertyTypes
-                    .map { property ->
-                        if (property.value.datatype == EdmPrimitiveTypeKind.Binary) {
-                            deletePropertyOfEntityFromS3(entitySetId, entityKeyIds, property.key)
-                        }
+        val numUpdates = hds.connection.use { connection ->
+            // Delete properties from S3
+            authorizedPropertyTypes.map { property ->
+                if (property.value.datatype == EdmPrimitiveTypeKind.Binary) {
+                    deletePropertyOfEntityFromS3(entitySetId, entityKeyIds, property.key)
+                }
+            }
 
-                        val ps = it.prepareStatement(deletePropertiesOfEntities(entitySetId, property.key))
-                        val arr = PostgresArrays.createUuidArray(it, entityKeyIds)
-                        ps.setArray(1, arr)
+            connection.autoCommit = false
+            val partitions = partitionsInfo.partitions.toList()
+            val propertyTypesArr = PostgresArrays.createUuidArray(connection, authorizedPropertyTypes.keys)
+            entityKeyIds
```

Comment:
it might be worth moving the group by outside of the connection just to avoid keeping it open for longer than absolutely necessary, but it would only matter in the case where there's a ton of entities being passed in

Reply:
:+1: please file ticket going to merge for now